backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -50,6 +147,41 @@ def update_configuration(event, context):
             e.response["ResponseMetadata"]["HTTPStatusCode"] = 429
         raise e
 
+    # Create the new app deny policy
+    try:
+        deny_policy_statements = []
+        enabled_services_dict = app_configuration.configuration.enabled_services.to_dict()
+        # Check each service
+        for service in enabled_services_dict.keys():
+            # If the service has permissions that need to be denied
+            if service in service_disable_permissions:
+                # If the service is deactivated
+                if not enabled_services_dict[service]:
+                    deny_policy_statements.extend(service_disable_permissions[service])
+
+        # Check for permissions associated with multiple services
+        for group in group_permissions:
+            # If all the services in a group are deactivated, then add the group deny statements to the deny policy
+            if "ServiceGroup" in group:
+                active_service_in_group = False
+                for group_service in group["ServiceGroup"]:
+                    if enabled_services_dict[group_service]:
+                        active_service_in_group = True
+                        break
+                # Add group to avoid duplicates
+                if not active_service_in_group and "Statements" in group:
+                    deny_policy_statements.extend(group["Statements"])
+
+        iam_manager.create_or_update_policy(
+            iam_manager.generate_policy_string(deny_policy_statements),
+            "app-denied-services",
+            "services",
+            "deny",
+            on_create_attach_to_notebook_role=True,
+        )
+    except Exception as e:
+        logging.error(f"Failed to create or update denied services policy {str(e)}")
```

Comment:
In what scenario would a failure here be acceptable? If we try to deny services and we fail to create the policy/attach it we need to communicate that to users. logging an error and then returning a success message means we'll have now way of letting users know that there was an issue.

Reply:
ACK: This is a draft. Working on figuring out what to do here. The config update should still be successful, so this is a mixed result. Looking to see if @api_wrapper has options for this.

Reply:
This is now addressed with a warning system that will attach warnings for any issues encountered, but will still notify that the app configuration update succeeded.